src/main/java/core/TaskList.java

```diff
@@ -0,0 +1,57 @@
+package core;
+
+import tasks.Task;
+
+import java.util.ArrayList;
+import java.util.stream.Stream;
+
+/**
+ * Controls the current list of tasks.
+ */
+public class TaskList {
+
+    private static final ArrayList<Task> tasks = new ArrayList<>();
```

Comment:
Perhaps 'tasks' should be entirely in uppercase since you marked it as a constant using 'final'?

Reply:
Below you have a function 'add' to add Task t to the ArrayList<Task> task, should the statement still have 'final' inside then since the array list will be changed? It may be less confusing to remove the term 'final'

Reply:
I don't agree that `tasks` is a constant simply because of the word `final`. As you have pointed out, the `ArrayList` is still mutable.

But I do want to keep the `final` modifier because I want to make it clear that `tasks` should not be assigned to.